arch/arc/mm/cache-arcv3.c

```diff
@@ -51,7 +52,9 @@ static int read_decode_cache_bcr_arcv3(int c, char *buf, int len)
 	if (cln0.has_scm) {
 		READ_BCR(ARC_REG_CLNR_SCM_BCR_0, scm0);
 
-		p_l2->sz_k = (1 << scm0.data_bank_sz) * (1 << scm0.data_banks);
+		assoc = (scm0.cache_assoc == 0) ? 16 : scm0.cache_assoc;
+		p_l2->sz_k = (scm0.cache_blk_sz << 6) * assoc *	(1 << scm0.cache_sets);
```

Comment:
I think you need to use the `L1_CACHE_SHIFT` definition instead of decimal-literal 6. The definition is formed from the ``CONFIG_ARC_CACHE_LINE_SHIFT`` option.

Reply:
You right, despite the fact that we are talking about L2 cache and not L1, I hope the line sizes are the same for L1 and L2

Reply:
The documentation states that the cache line size for ARСv3 is fixed and the size is 64 bytes. Let's rely on this and also fix the size = 64 bytes in code.